packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -133,11 +133,11 @@ function BlockPatternList( {
 			className="block-editor-block-patterns-list"
 			aria-label={ label }
 		>
-			{ blockPatterns.map( ( pattern ) => {
+			{ blockPatterns.map( ( pattern, index ) => {
 				const isShown = shownPatterns.includes( pattern );
 				return isShown ? (
 					<BlockPattern
-						key={ pattern.name }
```

Comment:
I don't understand, I thought pattern names are meant to be unique?

Reply:
I don't know why the pattern names are duplicated, but am I correct in assuming that the pattern name duplication itself should be corrected so that it does not happen?

Reply:
Yes, I think we should ensure that each pattern name gets a unique name otherwise we risk bigger issues (like when using patterns in themes where we only refer to them using the names).

Reply:
I have tried to fix this with 25d6d8c, but this does not seem to be enough as the test fails. I would like to close this PR and submit an issue, as it seems to be too difficult for me to fix this now 😅

Reply:
these patterns are the new `unsynced` `wp_block` CPT patterns added by users, so as with the existing reusable blocks they don't need to have a unique title as they are identified by the entity id. I will take a look at how we can resolve this issue.